gio/src/subclass/loadable_icon.rs

```diff
@@ -0,0 +1,263 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use std::{future::Future, pin::Pin};
+
+use crate::{
+    subclass::prelude::*, AsyncResult, Cancellable, GioFuture, InputStream, LoadableIcon, LocalTask,
+};
+use glib::{prelude::*, translate::*, GString};
+
+pub trait LoadableIconImpl: IconImpl {
+    fn load(
+        &self,
+        size: i32,
+        cancellable: Option<&Cancellable>,
+    ) -> Result<(InputStream, Option<GString>), glib::Error> {
+        self.parent_load(size, cancellable)
+    }
+
+    fn load_future(
+        &self,
+        size: i32,
+    ) -> Pin<Box<dyn Future<Output = Result<(InputStream, Option<GString>), glib::Error>> + 'static>>
+    {
+        self.parent_load_future(size)
+    }
+}
+
+pub trait LoadableIconImplExt: ObjectSubclass {
+    fn parent_load(
+        &self,
+        size: i32,
```

Comment:
Should this be an `IconSize`?

Reply:
It cannot be, IconSize is a gtk only enum